src/NServiceBus.Core/Reliability/Outbox/OutboxMessage.cs

```diff
@@ -26,19 +36,6 @@ public OutboxMessage(string messageId)
         /// <summary>
         ///     The list of operations performed during the processing of the incoming message.
         /// </summary>
-        public List<TransportOperation> TransportOperations
-        {
-            get
-            {
-                if (transportOperations == null)
-                {
-                    transportOperations = new List<TransportOperation>();
-                }
-
-                return transportOperations;
-            }
-        }
-
-        List<TransportOperation> transportOperations;
+        public IEnumerable<TransportOperation> TransportOperations { get; private set; }
```

Comment:
:+1: or a readonly collection although they are quirky

On Wednesday, September 30, 2015, Simon Cropp <notifications@github.com>
wrote:

> In src/NServiceBus.Core/Reliability/Outbox/OutboxMessage.cs
> <https://github.com/Particular/NServiceBus/pull/2912#discussion_r40739701>
> :
>
> > @@ -26,19 +36,6 @@ public OutboxMessage(string messageId)
> >          /// <summary>
> >          ///     The list of operations performed during the processing of the incoming message.
> >          /// </summary>
> > -        public List<TransportOperation> TransportOperations
> > -        {
> > -            get
> > -            {
> > -                if (transportOperations == null)
> > -                {
> > -                    transportOperations = new List<TransportOperation>();
> > -                }
> > -
> > -                return transportOperations;
> > -            }
> > -        }
> > -
> > -        List<TransportOperation> transportOperations;
> > +        public IEnumerable<TransportOperation> TransportOperations { get; private set; }
>
> since multiple operations are performed on this property it is a very
> dangerous API. depending on the specific IEnumerable returned we risk
> consuming the enumerable using things like .Count(). this should be
> changed back to a List
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Particular/NServiceBus/pull/2912/files#r40739701>.
>


-- 
*#NSBcon 2015 • The "All About NServiceBus" conference is back *
*• Learn more <http://go.particular.net/l/27442/2015-08-21/5d4wzp> →*